modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaLayoutServicesCheck.java

```diff
@@ -0,0 +1,174 @@
+		};
+
+		if (!hasParameterTypes(
+				content, content, ArrayUtil.toStringArray(parameterList),
+				parameterTypes)) {
+
+			addMessage(
+				content,
+				StringBundler.concat(
+					"Could not resolve types of addLayout method. The method ",
+					"signature has changed to addLayout(long userId, long ",
+					"groupId, boolean privateLayout, long parentLayoutId, long",
+					"classNameId, long classPK, Map<java.util.Locale, String> ",
+					"nameMap, Map<java.util.Locale, String> titleMap, ",
+					"Map<java.util.Locale, String> descriptionMap, ",
+					"Map<java.util.Locale, String> keywordsMap, ",
+					"Map<java.util.Locale, String> robotsMap, String type, ",
+					"String typeSettings, boolean hidden, boolean system, ",
+					"Map<java.util.Locale, String> friendlyURLMap, long ",
+					"masterLayoutPlid, ServiceContext serviceContext). Fill ",
+					"the new parameters manually."));
+		}
+
+		String newMethod = addMethodNewParameters(
+			parameterList.size() - 1, methodCall, ".addLayout(",
+			Arrays.asList("0"), parameterList);
+
+		return StringUtil.replace(content, methodCall, newMethod);
+	}
+
+	private String _replaceUpdateLayout(String content, String methodCall) {
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		if (parameterList.size() != 15) {
+			return content;
+		}
+
+		String[] parameterTypes = {
+			"long", "boolean", "long", "long", "Map<java.util.Locale, String>",
+			"Map<java.util.Locale, String>", "Map<java.util.Locale, String>",
+			"Map<java.util.Locale, String>", "Map<java.util.Locale, String>",
+			"String", "boolean", "Map<java.util.Locale, String>", "boolean",
+			"byte[]", "ServiceContext"
+		};
+
+		if (!hasParameterTypes(
+				content, content, ArrayUtil.toStringArray(parameterList),
+				parameterTypes)) {
+
+			addMessage(
+				content,
+				StringBundler.concat(
+					"Could not resolve types of updateLayout method. The ",
+					"method signature has changed to updateLayout(",
+					"long groupId, boolean privateLayout, long layoutId, long ",
+					"parentLayoutId, Map<java.util.Locale, String> nameMap, ",
+					"Map<java.util.Locale, String> titleMap, ",
+					"Map<java.util.Locale, String> descriptionMap, ",
+					"Map<java.util.Locale, String> keywordsMap, ",
+					"Map<java.util.Locale, String> robotsMap, String type, ",
+					"boolean hidden, Map<java.util.Locale, String> ",
+					"friendlyURLMap, boolean hasIconImage, byte[] iconBytes, ",
+					"long styleBookEntryId, long faviconFileEntryId, long ",
+					"masterLayoutPlid, ServiceContext serviceContext). Fill ",
+					"the new parameters manually."));
+		}
+
+		String newMethod = addMethodNewParameters(
+			parameterList.size() - 1, methodCall, ".updateLayout(",
+			Arrays.asList("0", "0", "0"), parameterList);
+
+		return StringUtil.replace(content, methodCall, newMethod);
+	}
+
+	private static final Pattern _addOrUpdateLayoutPattern = Pattern.compile(
+		"\\t*?\\w+(?:\\.updateLayout\\(|\\.addLayout\\()");
+
```

Comment:
@TamyrisTorres

-  you don't need the `?` before a `*`, the * means 0-any occurrences so the optional is already covered.

- If you escape the `.` before the conditional you don't need to repeat it. The same for the last `(`

`"\\t*\\w+\\.(?:updateLayout|addLayout)\\("`